cpp/tests/test_sde_sirs.cpp

```diff
@@ -0,0 +1,181 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "abm_helpers.h"
+#include "memilio/math/euler.h"
+#include "sde_sirs/model.h"
+#include "sde_sirs/simulation.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+const mio::ssirs::Model& ssirs_testing_model()
+{
+    static mio::ssirs::Model model;
+    model.step_size = 0.25;
+    model.populations.array().setConstant(1);
+    model.parameters.set<mio::ssirs::TimeImmune>(4);
+    { // set parameters s.t. coeffStoI is 1
+        model.parameters.set<mio::ssirs::TimeInfected>(1);
+        model.parameters.set<mio::ssirs::TransmissionProbabilityOnContact>(3);
+        model.parameters.get<mio::ssirs::ContactPatterns>().get_baseline()(0, 0) = 1;
+    }
+    return model;
+}
+
+TEST(TestSdeSirs, get_flows)
+{
+    // make two get_flows computations with mocked rng
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(6))
+        .WillOnce(testing::Return(1.))
+        .WillOnce(testing::Return(0.))
+        .WillOnce(testing::Return(1.))
+        .WillOnce(testing::Return(0.))
+        .WillOnce(testing::Return(1.))
+        .WillOnce(testing::Return(0.));
+
+    const Eigen::Vector3d y = Eigen::Vector3d::Constant(1);
+    Eigen::Vector3d flows   = Eigen::Vector3d::Constant(1);
+
+    // results contain two parts : deterministic + stochastic
+
+    ssirs_testing_model().get_flows(y, y, 0, flows);
+    auto expected_result = Eigen::Vector3d{1 + 2, 1, 0.25 + 1};
+    EXPECT_EQ(flows, expected_result);
+
+    ssirs_testing_model().get_flows(y, y, 0, flows);
+    expected_result = Eigen::Vector3d{1, 1 + 2, 0.25};
+    EXPECT_EQ(flows, expected_result);
+}
+
+TEST(TestSdeSirs, Simulation)
+{
+    // make a single integration step via a simulation
+    // this should overwrite the model step size
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(3))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5));
+
+    auto integrator = std::make_shared<mio::EulerIntegratorCore>();
+    auto sim        = mio::ssirs::Simulation(ssirs_testing_model(), 0, 1);
+    sim.set_integrator(integrator);
+    sim.advance(1);
+
+    EXPECT_EQ(sim.get_model().step_size, 1.0); // set by simulation
```

Comment:
Don't understand why this is 1.0. Isn't the default 0.1? In the `model` and `Simulation`?